doc/user/tutorials/making_your_first_app.rst

```diff
@@ -435,76 +441,4 @@ to bring the ARMI state back to any state point from the run for analysis.
 
 A generic description of the outputs is provided in :doc:`/user/outputs/index`. 
 
-Adding new output
------------------
-Besides adding new state variables to the ARMI database, your plugins and applications may
-also find it beneficial to generate new output files, such as plots or other reports.
-Let's add automatic plots of coolant temperature at runtime in our application. 
-
-We can use the :ref:`detail assembly feature <detail-assems>` to allow users to
-choose which assembly will get printed out, since you may not want plots for every
-assembly.
-
-Open up the ``thermalSolver.py`` plugin code and add the following function to plot a
-single assembly:
-
-.. code-block:: python
-    :caption: ``myapp/thermalSolver.py``
-
-    import matplotlib.pyplot as plt
-
-    def plotOutput(a, fname):
-        """Make a temperature vs. height plot for a single assem"""
-        heights = []
-        temperatures = []
-        height = 0.0 
-
-        for b in a:
-            heights.append(height)
-            temperatures.append(b.p.THcoolantAverageT)
-            height += b.getHeight()
-
-        plt.figure()
-        plt.plot(heights, temperatures, "-o")
-        plt.title(f"Temperatures for {a}")
-        plt.xlabel("Axial height (cm)")
-        plt.ylabel("Coolant temperature (°C)")
-        plt.savefig(fname)
-        plt.close()
-
-To get the code to run on the detail assemblies, we change the ``interactEveryNode`` hook
-to call the plot function, as follows: 
-
-.. code-block:: python
-    :caption: ``myapp/thermalSolver.py``
-
-    def interactEveryNode(self, cycle=None, timeNode=None):
-        runLog.info("Computing idealized flow rate")
-        for assembly in self.r.core:
-            runThermalHydraulics(assembly)
-
-        history = self.getInterface("history")
-        for a in history.getDetailAssemblies():
```

Comment:
Can we keep this plotting stuff, but just replace these two lines with a `getLocationContents()` or similar?